nntrainer/tensor/blas_neon.cpp

```diff
@@ -701,6 +702,188 @@ void sgemv_neon_fp16(const __fp16 *A, const __fp16 *X, __fp16 *Y, uint32_t rows,
+  }
+  if (rows / 16 >= batch) {
+    for (unsigned int i = 0; i < rows; i += 16) {
+      __fp16 x = alpha * (X[i]);
+      __fp16 x2 = alpha * (X[i + 1]);
+      __fp16 x3 = alpha * (X[i + 2]);
+      __fp16 x4 = alpha * (X[i + 3]);
+      __fp16 x5 = alpha * (X[i + 4]);
+      __fp16 x6 = alpha * (X[i + 5]);
+      __fp16 x7 = alpha * (X[i + 6]);
+      __fp16 x8 = alpha * (X[i + 7]);
+      __fp16 x9 = alpha * (X[i + 8]);
+      __fp16 x10 = alpha * (X[i + 9]);
+      __fp16 x11 = alpha * (X[i + 10]);
+      __fp16 x12 = alpha * (X[i + 11]);
+      __fp16 x13 = alpha * (X[i + 12]);
+      __fp16 x14 = alpha * (X[i + 13]);
+      __fp16 x15 = alpha * (X[i + 14]);
+      __fp16 x16 = alpha * (X[i + 15]);
+#pragma omp parallel for schedule(guided) num_threads(NEON_NUM_THREADS)
+      for (int idx = 0; idx < cols; idx += 8) {
+        float16x8_t wvec0_7_f16 = vmulq_n_f16(vld1q_f16(&A[i * cols + idx]), x);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 1) * cols + idx]), x2);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 2) * cols + idx]), x3);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 3) * cols + idx]), x4);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 4) * cols + idx]), x5);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 5) * cols + idx]), x6);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 6) * cols + idx]), x7);
+        wvec0_7_f16 =
+          vfmaq_n_f16(wvec0_7_f16, vld1q_f16(&A[(i + 7) * cols + idx]), x8);
+
+        float16x8_t w2vec0_7_f16 =
+          vmulq_n_f16(vld1q_f16(&A[(i + 8) * cols + idx]), x9);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 9) * cols + idx]), x10);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 10) * cols + idx]), x11);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 11) * cols + idx]), x12);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 12) * cols + idx]), x13);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 13) * cols + idx]), x14);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 14) * cols + idx]), x15);
+        w2vec0_7_f16 =
+          vfmaq_n_f16(w2vec0_7_f16, vld1q_f16(&A[(i + 15) * cols + idx]), x16);
+
+        float32x4_t y0_3 = vaddq_f32(vld1q_f32(&Y32[idx]),
+                                     vcvt_f32_f16(vget_low_f16(wvec0_7_f16)));
+        y0_3 = vaddq_f32(y0_3, vcvt_f32_f16(vget_low_f16(w2vec0_7_f16)));
+        float32x4_t y4_7 = vaddq_f32(vld1q_f32(&Y32[idx + 4]),
+                                     vcvt_f32_f16(vget_high_f16(wvec0_7_f16)));
+        y4_7 = vaddq_f32(y4_7, vcvt_f32_f16(vget_high_f16(w2vec0_7_f16)));
+
+        vst1q_f32(&Y32[idx], y0_3);
+        vst1q_f32(&Y32[idx + 4], y4_7);
```

Comment:
I found most part of the code in `sgemv_transpose_neon_fp16_multithread` is exact same as in `sgemv_transpose_neon_fp16_general`.

For instance,
https://github.com/nnstreamer/nntrainer/blob/0bfbf34e1a50d53bce04c57bf146e4ff5010d052/nntrainer/tensor/blas_neon.cpp#L746-L787

this is a long code and I'm pretty sure there's room for improvement to minimize these codes. 
(e.g., combine it into a single function or utilize helper functions)